appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java

```diff
@@ -1679,7 +1673,18 @@ private void deployAll() throws GlassFishException {
         LOGGER.log(Level.INFO, "Deployed {0} archive(s)", deploymentCount);
     }
 
-    private boolean hasJavaArchiveExtension(String filePath) {
+    private String removeJavaArchiveExtension(String fileName) {
+        if (hasJavaArchiveExtension(fileName)) {
+            return fileName.substring(0, fileName.length() - 4);
+        } else {
+            return fileName;
+        }
+    }
+
+    private static boolean hasJavaArchiveExtension(String filePath) {
```

Comment:
I'm probably being blind, but is there a reason this is being made static?

Reply:
Because it has no access to fields. Good enough reason for me :)